complex/cargo/list/src/lib.rs

```diff
@@ -224,6 +225,44 @@ impl ListManager {
         uuids
     }
 
+    pub fn fetch_all_items(&self) -> Vec<ItemJNA> {
```

Comment:
Take a look at the `Into` trait. https://doc.rust-lang.org/std/convert/trait.Into.html

This is a trait you can implement on structs to convert one thing into another. Think about implementing Into for `Item` to `ItemJNA` and `Vec<Item>` to `Vec<ItemJNA>`. That way the `ListManager` function can deal entirely with safe types and code and you can call `list.into()` from inside your FFI function when `Box`ing it for return.

Reply:
What's nice about `Into` is that implementing `Into` automatically implements `From` which means when we pass an `ItemJNA` back into we can call `item.from()` to turn it back into an `Item`.